src/raft/raw_node.rs

```diff
@@ -124,6 +131,9 @@ impl Ready {
         if raft.raft_log.get_unstable().snapshot.is_some() {
             rd.snapshot = raft.raft_log.get_unstable().snapshot.clone().unwrap();
         }
+        if !raft.read_states.is_empty() {
+            rd.read_states = raft.read_states.drain(..).collect();
```

Comment:
These read states in `raft.read_states` would be ready to read out, the pending ones are in read only struct.